### PR TITLE
SoC: add some paragraphs about Git development process

### PR DESCRIPTION
**Pull request by chriscool**
I think the following on top of your changes might help too.

### CODE REVIEW

File: SoC-2016-Ideas.md [hunk 1]
```diff
@@ -46,20 +46,34 @@ your microproject in your application (e.g. linking to
...
+put "[RFC/PATCH]" at the beginning of its subject. If you are not sure
+about your proposal, you can discuss that in the same email where you
+introduce yourself or in separate emails. Please use "[GSoC]" at the
+beginning of such emails.
 
 In summary, all applicants must (not necessarily in this order):
 
-* Complete a microproject.
+* Complete a microproject. Send the corresponding patch(s) early and
+  often to the mailing list, responding to reviews and comments.
```
Comment (moy) [inline SoC-2016-Ideas.md:73]: I think it's better to keep this item short: the microproject page already contains a lot of information about how to proceed, and the point of a summary is to remain as short as possible IMHO.

File: SoC-2016-Ideas.md [hunk 2]
```diff
@@ -46,20 +46,34 @@ your microproject in your application (e.g. linking to
...
 
-* Write a detailed application explaining their project.
-
-* Discuss their project by posting drafts of their application on the
-  mailing-list long before the deadline.
+* Write a detailed application explaining their project. Send it early
+  and often to the mailing list, responding to reviews and comments.
```
Comment (moy) [inline SoC-2016-Ideas.md:76]: I'm not convinced by the merge of two items: I made them separate to insist on the fact that "write a detailed application" was not sufficient, and that the "discuss" part was also mandatory. I see the list as a checklist: if one item was not done, the application is not acceptable.

File: SoC-2016-Ideas.md [hunk 3]
```diff
@@ -46,20 +46,34 @@ your microproject in your application (e.g. linking to
...
+patch series to the mailing list until they are considered good and
+correct by the reviewers and Junio Hamano, the maintainer, who will
+merge them. This process usually takes quite some time. By sending
+drafts of your proposal, and patches for your microproject, to the
+mailing list long before the deadline, you can show us that you are
+willing and able to work well using the Git development process.
+
+It is *expected* that what you send will need several rounds of
+reviews and discussions. If you not sure at all about a patch you can
+put "[RFC/PATCH]" at the beginning of its subject. If you are not sure
```
Comment (moy) [inline SoC-2016-Ideas.md:65]: This is partly redundant with the microproject page which already talks about the flow, but provides useful additions. I'm wondering if we shouldn't instead add the above to the microproject page (near the link to SubmittingPatches) and write something like this here:

"Just like for submitting patches (see the explanations in the [microproject](SoC-2016-Microprojects) page), your proposal is likely to trigger a lot of discussion, and it is expected that you send multiple iterations of your proposal."

Comment (chriscool) [inline SoC-2016-Ideas.md:65]: Yeah, I am ok with this.

File: SoC-2016-Ideas.md (hunk 1 above)
Comment (chriscool) [inline SoC-2016-Ideas.md:73]: Ok, maybe add the link to the microproject here too.

File: SoC-2016-Ideas.md (hunk 2 above)
Comment (chriscool) [inline SoC-2016-Ideas.md:76]: Ok then.

**Comment by chriscool** [comment]
In the above patch there is a check list for what should be in the application.

File: SoC-2016-Ideas.md [hunk 4]
```diff
@@ -75,6 +75,27 @@ In summary, all applicants must (not necessarily in this order):
...
+* Include link(s) to the mailing-list discussion(s) related to your
+  microproject.
+
+* Include what is suggested in
+  [the GSoC Student Guide](http://write.flossmanuals.net/gsocstudentguide/writing-a-proposal/)
+
+([gmane](http://news.gmane.org/gmane.comp.version-control.git) can be
+used for searching the mailing list and linking to previous
+discussions.)
+
```
Comment (moy) [inline SoC-2016-Ideas.md:98]: Very good. Thanks!

File: SoC-2016-Ideas.md [hunk 5]
```diff
@@ -75,6 +75,27 @@ In summary, all applicants must (not necessarily in this order):
 * Write a detailed application explaining their project. Send it early
   and often to the mailing list, responding to reviews and comments.
 
+In your application, it is a good idea to:
+
+* Include link(s) to the mailing-list discussion(s) related to the
+  project you chose in your application, for example previous
+  discussions or patch series about the topic. There might be
+  interesting discussions about the topics that are several year old.
+  It is also a good idea to summerize them.
```
Comment (moy) [inline SoC-2016-Ideas.md:84]: Typo: summarize.

Comment (chriscool) [inline SoC-2016-Ideas.md:84]: Ok, thanks!

**Comment by moy** [comment]
I incorporated your changes adding my remarks/fixes. Feel free to amend if I didn't do it properly or if you disagree with the way I did it. Thanks,

**Comment by chriscool** [comment]
It looks great, thanks!

**Pull request closed without merging**